ext/openssl/ossl_pkey_ec.c

```diff
@@ -400,13 +400,61 @@ static VALUE ossl_ec_key_is_private(VALUE self)
 
 /*
  *  call-seq:
- *     key.export([cipher, pass_phrase]) => String
- *     key.to_pem([cipher, pass_phrase]) => String
+ *     key.export([cipher, password]) => String
+ *     key.to_pem([cipher, password]) => String
```

Comment:
If you replace the variable `pass_phrase` with the `password` here, do you like to replace the following parts too?

```
$ grep -r pass_phrase lib/ ext/
ext/openssl/ossl.c: *   pass_phrase = 'my secure pass phrase goes here'
ext/openssl/ossl.c: *   key_secure = key.private_to_pem cipher, pass_phrase
ext/openssl/ossl.c: *   pass_phrase = 'my secure pass phrase goes here'
ext/openssl/ossl.c: *   key4 = OpenSSL::PKey.read key4_pem, pass_phrase
ext/openssl/ossl.c: *   pass_phrase = 'my secure pass phrase goes here'
ext/openssl/ossl.c: *   encryptor.pkcs5_keyivgen pass_phrase, salt
ext/openssl/ossl.c: *   decryptor.pkcs5_keyivgen pass_phrase, salt
ext/openssl/ossl.c: *   pass_phrase = 'my secure pass phrase goes here'
ext/openssl/ossl.c: *     io.write ca_key.private_to_pem(cipher, pass_phrase)
```

Reply:
Good point. To my understanding, the words password and /pass.?phrase/ are completely interchangeable, but they haven't been used consistently.

In fact, RFCs and OpenSSL man pages are inconsistent in word choice.

 - RFC 8018: PKCS#5 (PBKDF2, `OpenSSL::KDF.pbkdf2_hmac`) which is used by PKCS#8's encryption uses the word "password".
 - RFC 7292: PKCS#12 (`OpenSSL::PKCS12`) also uses "password".
 - RFC 7914: scrypt (`OpenSSL::KDF.scrypt`) describes itself as a "password-based key derivation function", while it names the input parameter as the "passphrase".
 - OpenSSL's man pages appear to prefer "passphrase" in recent years, but OpenSSL has both functions named "password" and "passphrase". Notably, OpenSSL 3.0.0 added two functions named `OSSL_ENCODER_CTX_set_pem_password_cb()` and `OSSL_ENCODER_CTX_set_passphrase_cb()` which appear to serve the same purpose (!?).

I will push a new change to replace all /pass.?phrase/ with "password", except scrypt where "passphrase" is more appropriate.

Reply:
Sure. Thanks for explaining it. OK. I see that you originally named the variables of the functions referring to the words used in the RFC documents.

* [RFC 8018: PKCS#5](https://www.rfc-editor.org/rfc/rfc8018) - 2. Notation - "P: password, an octet string"
* [RFC 7914: scrypt ](https://www.rfc-editor.org/rfc/rfc7914) - 6. The scrypt Algorithm - "P: Passphrase, an octet string."

> I will push a new change to replace all /pass.?phrase/ with "password", except scrypt where "passphrase" is more appropriate.

It sounds good to me!

Reply:
I replaced the occurrences of "passphrase" with "password" except those in the OpenSSL::Cipher#pkcs5_keyivgen example, which I started to think having the example itself does harm (#647):

> ext/openssl/ossl.c: *   encryptor.pkcs5_keyivgen pass_phrase, salt
> ext/openssl/ossl.c: *   decryptor.pkcs5_keyivgen pass_phrase, salt